### PR TITLE
Django 4.2 upgrade

### PR DESCRIPTION
**Pull request by salman2013**
**Description:**

- Add support for  Django 4.2

**Codemod used to upgrade this repo:**  https://github.com/adamchainz/django-upgrade

**Ticket:** https://github.com/edx/upgrades/issues/142

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @salman2013!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/openedx/.github/blob/master/CONTRIBUTING.md)     file for more information.     If you've signed an agreement in the past, you may need to re-sign.  See     [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/)     for details.

    
Once you've signed the CLA, please allow 1 business day for it to be processed.  After this time, you can re-run the     CLA check by editing the PR title.  If the problem persists, you can tag the `@openedx/cla-problems` team in a     comment on your PR for further assistance.

File: edx_proctoring/__init__.py [hunk 1]
```diff
@@ -4,5 +4,3 @@
 
 # Be sure to update the version number in edx_proctoring/package.json
 __version__ = '4.15.1'
```
Comment (UsamaSadiq) [inline edx_proctoring/__init__.py:6]: Let's bump the version as well. 
```suggestion
__version__ = '4.16.0'
```

File: edx_proctoring/admin.py [hunk 2]
```diff
@@ -43,6 +43,7 @@ class Meta:
         fields = '__all__'
 
 
+@admin.register(ProctoredExam)
```
Comment (UsamaSadiq) [inline edx_proctoring/admin.py:46]: Please link the release notes for this change.

File: tox.ini [hunk 3]
```diff
@@ -1,80 +1,84 @@
...
+    pii_check,
+    translations
+
 [testenv]
-deps =
+deps = 
     django32: Django>=3.2,<3.3
     django40: Django>=4.0,<4.1
+    django42: Django>=4.2,<4.3
+    django40: Django>=4.0,<4.1
```
Comment (UsamaSadiq) [inline tox.ini:14]: Let's remove the duplicate entry
```suggestion
```

**Review by UsamaSadiq — CHANGES_REQUESTED** [review]

File: edx_proctoring/__init__.py (hunk 1 above)
Comment (salman2013) [reply edx_proctoring/__init__.py]: done with comment.

File: tox.ini [hunk 4]
```diff
@@ -1,80 +1,83 @@
...
-        quality,
-        version_check,
-        pii_check,
-        translations
+envlist = 
+    py38-celery{50}-django{32,40,42}-drflatest
+    quality,
+    version_check,
+    pii_check,
+    translations
```
Comment (UsamaSadiq) [inline tox.ini:7]: Let's try running translations check with Django 42 as well. 
```suggestion
    translations-django{32, 42}
```

File: tox.ini [hunk 5]
```diff
@@ -1,80 +1,83 @@
...
-commands =
+commands = 
     make pii_check
 
 [testenv:translations]
-whitelist_externals =
+whitelist_externals = 
     make
-deps =
+deps =
```
Comment (UsamaSadiq) [inline tox.ini:77]: Update deps to install custom django version for testing 
```
    django32: Django>=3.2,<3.3
    django42: Django>=4.2,<4.3
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -22,6 +22,7 @@ jobs:
           [
             "django32-drflatest",
             "django40-drflatest",
+            "django42-drflatest",
```
Comment (UsamaSadiq) [inline .github/workflows/ci.yml:25]: also update `translations` check to run with both `django32` and `django42`. 
e.g. 
```
translations-django32,
translations-django42
```

**Review by UsamaSadiq — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 6 above)
Comment (salman2013) [reply .github/workflows/ci.yml:25]: done with comment.

File: tox.ini (hunk 5 above)
Comment (salman2013) [reply tox.ini]: done with comment.

File: tox.ini (hunk 4 above)
Comment (salman2013) [reply tox.ini]: done with comment.

**Review by UsamaSadiq — APPROVED** [review]

**Review by awais786 — APPROVED** [review]

**Pull request merged**